### PR TITLE
Create PURLs for URLs that don't have a router #68

### PR DESCRIPTION
**Pull request by JonoYang**
I am updating `url2purl` to create a generic PackageURL whenever it encounters a URL that does not have a route available.

### CODE REVIEW

File: src/packageurl/contrib/url2purl.py [hunk 1]
```diff
@@ -53,7 +53,9 @@ def url2purl(url):
         try:
             return purl_router.process(url)
         except NoRouteAvailable:
-            return
+            # If `url` does not fit in one of the existing routes,
+            # we attempt to create a generic PackageURL for `url`
+            return build_generic_purl(url)
```
Comment (JonoYang) [inline src/packageurl/contrib/url2purl.py:58]: @pombredanne I am not sure if this is the best place to create a generic purl for a given URL if there is no other route that handles the URL.

Comment (pombredanne) [reply src/packageurl/contrib/url2purl.py:58]: I think this makes perfect sense to have this here :+1:

**Review by pombredanne — APPROVED** [review]
Neat and clean. Thank you++

File: tests/contrib/data/url2purl.json [hunk 2]
```diff
@@ -234,21 +226,20 @@
...
-  "https://bitbucket.org/robeden/trove/downloads/trove-3.0.3.zip" : "pkg:bitbucket/robeden/trove?download_url=https://bitbucket.org/robeden/trove/downloads/trove-3.0.3.zip",
-
+  "https://bitbucket.org/multicoreware/x265/downloads/x265_2.6.tar.gz": "pkg:bitbucket/multicoreware/x265?download_url=https://bitbucket.org/multicoreware/x265/downloads/x265_2.6.tar.gz",
+  "https://bitbucket.org/robeden/trove/downloads/trove-3.0.3.zip": "pkg:bitbucket/robeden/trove?download_url=https://bitbucket.org/robeden/trove/downloads/trove-3.0.3.zip",
   "https://gitlab.com/TG1999/firebase/-/tree/1a122122/views": "pkg:gitlab/tg1999/firebase@1a122122#views",
   "https://gitlab.com/tg1999/firebase": "pkg:gitlab/tg1999/firebase",
   "https://gitlab.com/TG1999/firebase/-/": "pkg:gitlab/tg1999/firebase",
   "https://gitlab.com/TG1999/firebase/-/tree": "pkg:gitlab/tg1999/firebase",
-  "https://gitlab.com/TG1999/firebase/-/master": "pkg:gitlab/tg1999/firebase@master",
+  "https://gitlab.com/TG1999/firebase/-/master": "pkg:gitlab/tg1999/firebase",
```
Comment (pombredanne) [inline tests/contrib/data/url2purl.json:235]: May this be a regression?

**Review by pombredanne — APPROVED** [review]
LGTM! Just a tiny note on one of the edge cases tests for github for your review.

File: tests/contrib/data/url2purl.json (hunk 2 above)
Comment (JonoYang) [reply tests/contrib/data/url2purl.json:235]: @pombredanne I think this is fine. I don't think `https://gitlab.com/TG1999/firebase/-/master` is a proper URL to a gitlab repo branch. 

I've tested this out by creating my own gitlab repo (https://gitlab.com/JonoYang/test) with a branch named `main`, then going to the URL `https://gitlab.com/JonoYang/test/-/main`. I get a 404 when I go to `https://gitlab.com/JonoYang/test/-/main`.

**Review by pombredanne — APPROVED** [review]
Thank for the updates. LGTM ... merging!

**Comment by pombredanne** [comment]
@JonoYang Do you want to joint as a committer?

**Comment by JonoYang** [comment]
@pombredanne That'd be great!

**Comment by pombredanne** [comment]
ack. Thanks. Doing it now
